grouping.js

```diff
@@ -174,28 +206,43 @@ Partitioner = {
 	_directOps: new Meteor.EnvironmentVariable(),
 	_searchAllUsers: new Meteor.EnvironmentVariable(),
 
-	setUserGroup(userId, groupId) {
+	async setUserGroupAsync(userId, groupId) {
 		check(userId, String);
 		check(groupId, String);
 
-		if (Meteor.users._partitionerDirect.findOne(userId, {fields: {group: 1}}).group) {
+		if (await Meteor.users._partitionerDirect.findOneAsync(userId, {fields: {group: 1}}).group) {
 			throw new Meteor.Error(403, "User is already in a group");
 		}
 
 		return Meteor.users._partitionerDirect.update(userId, {$set: {group: groupId}});
 	},
 
+	// Static method to backwards-compatibly
+	setUserGroup(userId, groupId) {
+		return Meteor.wrapAsync(this.setUserGroupAsync)(userId, groupId);
+	},
+
+	async getUserGroupAsync(userId) {
+		check(userId, String);
+		return (await Meteor.users._partitionerDirect.findOneAsync(userId, {fields: {group: 1}}) || {}).group;
+	},
+
+	// Static method to backwards-compatibly
 	getUserGroup(userId) {
+		return Meteor.wrapAsync(this.getUserGroupAsync)(userId);
```

Comment:
@wildhart here I don't know why wrapAsync doesn't works and still return Promise instead of groupId